packages/x-date-pickers/src/tests/describeGregorianAdapter/testCalculations.ts

```diff
@@ -164,122 +277,343 @@ export const testCalculations: DescribeGregorianAdapterTestSuite = ({ adapter })
+    });
+
+    it('should work with different timezones', function test() {
+      if (!adapter.isTimezoneCompatible) {
+        this.skip();
+      }
+
+      // Both dates below have the same timestamp, but they are not in the same year when represented in their respective timezone.
+      // The adapter should still consider that they are in the same year.
+      const londonTimezone = adapterTZ.endOfYear(
+        adapterTZ.setTimezone(testDateIso, 'Europe/London'),
+      );
+      const parisTimezone = adapterTZ.setTimezone(londonTimezone, 'Europe/Paris');
 
-    const nextYearLocale = adapter.addYears(testDateLocale, 1);
-    expect(adapter.isAfterYear(nextYearLocale, testDateLocale)).to.equal(true);
-    expect(adapter.isAfterYear(testDateLocale, nextYearLocale)).to.equal(false);
+      expect(adapterTZ.isAfterYear(londonTimezone, parisTimezone)).to.equal(false);
+      expect(adapterTZ.isAfterYear(parisTimezone, londonTimezone)).to.equal(false);
+    });
   });
 
-  it('Method: isAfterDay', () => {
-    const nextDayIso = adapter.addDays(testDateIso, 1);
-    expect(adapter.isAfterDay(nextDayIso, testDateIso)).to.equal(true);
-    expect(adapter.isAfterDay(testDateIso, nextDayIso)).to.equal(false);
+  describe('Method: isAfterDay', () => {
+    it('should work with the same timezone', () => {
+      const nextDayIso = adapter.addDays(testDateIso, 1);
+      expect(adapter.isAfterDay(nextDayIso, testDateIso)).to.equal(true);
+      expect(adapter.isAfterDay(testDateIso, nextDayIso)).to.equal(false);
+
+      const nextDayLocale = adapter.addDays(testDateLocale, 1);
+      expect(adapter.isAfterDay(nextDayLocale, testDateLocale)).to.equal(true);
+      expect(adapter.isAfterDay(testDateLocale, nextDayLocale)).to.equal(false);
+    });
+
+    it('should work with different timezones', function test() {
+      if (!adapter.isTimezoneCompatible) {
+        this.skip();
+      }
+
+      // Both dates below have the same timestamp, but they are not in the same day when represented in their respective timezone.
+      // The adapter should still consider that they are in the same day.
+      const londonTimezone = adapterTZ.endOfDay(
+        adapterTZ.setTimezone(testDateIso, 'Europe/London'),
+      );
+      const parisTimezone = adapterTZ.setTimezone(londonTimezone, 'Europe/Paris');
+
+      expect(adapterTZ.isAfterDay(londonTimezone, parisTimezone)).to.equal(false);
+      expect(adapterTZ.isAfterDay(parisTimezone, londonTimezone)).to.equal(false);
 
-    const nextDayLocale = adapter.addDays(testDateLocale, 1);
-    expect(adapter.isAfterDay(nextDayLocale, testDateLocale)).to.equal(true);
-    expect(adapter.isAfterDay(testDateLocale, nextDayLocale)).to.equal(false);
+      // Both dates below have the same day when represented in their respective timezone,
+      // But not when represented in the same timezone
+      // The adapter should consider that they are not in the same day
```

Comment:
Are they? 🤔 
It looks like you are adding `-1` hour to one of the value 🤔

Reply:
Could you elaborate?
For me the comment is correct.
When represented in their respective timezone, the dates are on the same day.
When represented in the timezone of the 1st date, the dates are not on the same day (because of the 1 hour gap).

Reply:
Nevermind. Looked at it with fresh eyes and it is correct. Sorry for the confusion 🙈 😊

Reply:
I can't think of a clear improvement to the comment ATM.
It's a complex topic anyways like you mentioned. 🙈